ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.h

```diff
@@ -0,0 +1,278 @@
+class PropertyVector;
+}
+
+namespace ProcessLib
+{
+class SpatialPosition;
+namespace TwoPhaseFlowWithPrho
+{
+class TwoPhaseFlowWithPrhoMaterialProperties
+{
+public:
+    static int const JacobianResidualSize = 2;
+    using ResidualVector = Eigen::Matrix<double, JacobianResidualSize, 1>;
+    using JacobianMatrix = Eigen::Matrix<double, JacobianResidualSize,
+                                         JacobianResidualSize, Eigen::RowMajor>;
+    using UnknownVector = Eigen::Matrix<double, JacobianResidualSize, 1>;
+
+public:
+    using ArrayType = MaterialLib::Fluid::FluidProperty::ArrayType;
+
+    TwoPhaseFlowWithPrhoMaterialProperties(
+        bool const has_material_ids,
+        MeshLib::PropertyVector<int> const& material_ids,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            liquid_density,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            viscosity,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            gas_density,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            gas_viscosity,
+        std::vector<Eigen::MatrixXd>
+            intrinsic_permeability_models,
+        std::vector<std::unique_ptr<MaterialLib::PorousMedium::Porosity>>&&
+            porosity_models,
+        std::vector<std::unique_ptr<MaterialLib::PorousMedium::Storage>>&&
+            storage_models,
+        std::vector<std::unique_ptr<
+            MaterialLib::PorousMedium::CapillaryPressureSaturation>>&&
+            capillary_pressure_models,
+        std::vector<
+            std::unique_ptr<MaterialLib::PorousMedium::RelativePermeability>>&&
+            relative_permeability_models);
+
+    void setMaterialID(const ProcessLib::SpatialPosition& pos);
+
+    Eigen::MatrixXd const& getPermeability(
+        const double t,
+        const ProcessLib::SpatialPosition& pos,
+        const int dim) const;
+
+    double getPorosity(const double t, const ProcessLib::SpatialPosition& pos,
+                       const double p, const double T,
+                       const double porosity_variable) const;
+
+    double getNonwetRelativePermeability(const double t,
+                                         const ProcessLib::SpatialPosition& pos,
+                                         const double p, const double T,
+                                         const double saturation) const;
+    double getWetRelativePermeability(const double t,
+                                      const ProcessLib::SpatialPosition& pos,
+                                      const double p, const double T,
+                                      const double saturation) const;
+    double getCapillaryPressure(const double t,
+                                const ProcessLib::SpatialPosition& pos,
+                                const double p, const double T,
+                                const double saturation) const;
+    double getDerivCapillaryPressure(const double t,
```

Comment:
I'd prefere `getCapillaryPressureDerivative`. The five more characters will not make any difference on coding speed, but improve the readabilitiy.